lib/Fencer/Types.hs

```diff
@@ -132,6 +133,15 @@ instance FromJSON RateLimit where
 -- Rate limit rule configs
 ----------------------------------------------------------------------------
 
+-- | A class describing how to access descriptor definitions within a
+-- type, if there are any present at all.
+--
+-- This class is needed for accessing descriptor definitions in a
+-- uniform way both when dealing with domain definitions and when
+-- dealing with descriptor definitions.
+class HasDescriptors a where
+  descriptorsOf :: a -> [DescriptorDefinition]
```

Comment:
Classes like that usually are not only getters, but also setters. I.e. it's usually

```
descriptors :: Lens' a [DescriptorDefinition]
```

But it's fine to have getter-only classes if you never need any setters.

Reply:
That's a good point.

I'll just use a meek self-defense argument and say we haven't brought in lenses as a dependency so far and that this small use case doesn't justify doing that. How about that? :smile:

Reply:
Quite reasonable :)